api/v1/users/PKPUserHandler.inc.php

```diff
@@ -15,92 +15,50 @@
-                    'pattern' => $this->getEndpointPattern(),
-                    'handler' => [$this, 'getMany'],
-                    'roles' => $roles
-                ],
-                [
-                    'pattern' => $this->getEndpointPattern() . '/reviewers',
-                    'handler' => [$this, 'getReviewers'],
-                    'roles' => $roles
-                ],
-                [
-                    'pattern' => $this->getEndpointPattern() . '/{userId:\d+}',
-                    'handler' => [$this, 'get'],
-                    'roles' => $roles
-                ],
-                [
-                    'pattern' => $this->getEndpointPattern() . '/report',
-                    'handler' => [$this, 'getReport'],
-                    'roles' => $roles
-                ],
-            ],
-        ];
-        parent::__construct();
-    }
-
-    /**
-     * @copydoc PKPHandler::authorize()
-     */
-    public function authorize($request, &$args, $roleAssignments)
-    {
-        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
-        return parent::authorize($request, $args, $roleAssignments);
-    }
+    use AuthorizesRequests;
+    use DispatchesJobs;
+    use ValidatesRequests;
 
     /**
      * Get a collection of users
-     *
-     * @param Request $slimRequest Slim request object
-     * @param Response $response object
-     * @param array $args arguments
-     *
-     * @return Response
      */
-    public function getMany($slimRequest, $response, $args)
+    public function getMany(Request $request): JsonResponse
     {
-        $request = $this->getRequest();
-        $context = $request->getContext();
-
-        if (!$context) {
-            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
-        }
+        $querystrings = $request->query(null);
+        $params = $this->processAllowedParams(
+            $querystrings,
+            [
+                'assignedToCategory',
+                'assignedToSection',
+                'assignedToSubmission',
+                'assignedToSubmissionStage',
+                'count',
+                'offset',
+                'orderBy',
+                'orderDirection',
+                'roleIds',
+                'searchPhrase',
+                'status',
+            ]
+        );
+
+        $params['contextId'] = $request->attributes->get('pkpContext')->getId();
 
-        $params = $this->_processAllowedParams($slimRequest->getQueryParams(), [
-            'assignedToCategory',
-            'assignedToSection',
-            'assignedToSubmission',
-            'assignedToSubmissionStage',
-            'count',
-            'offset',
-            'orderBy',
-            'orderDirection',
-            'roleIds',
-            'searchPhrase',
-            'status',
-        ]);
-
-        $params['contextId'] = $context->getId();
-
-        HookRegistry::call('API::users::params', [&$params, $slimRequest]);
```

Comment:
Rather than removing the hook entirely, I would suggest:
- Providing an equivalent set of parameters to it (`$params, $request`)
- Documenting the change in conventions at the top of the PR, so Nate can be flagged about developer documentation updates when this is merged.

(Same with others below.)

If the hook is removed, it'll make it really tough for developers who use the hook to debug why -- their code will stop getting called and they won't know why. A breaking change (the slim request being swapped out for a Laravel request object) will give them an error message they will be able to work with.